scripts/process.py

```diff
@@ -0,0 +1,161 @@
+
+    Stock API are come and go.
+    Presently, this uses the IEX Trading API by IEX Group, Inc.
+    Data provided for free by IEX. https://iextrading.com/developer
+"""
+
+from dataflows import Flow, PackageWrapper, ResourceWrapper, validate
+from dataflows import add_metadata, dump_to_path, load, set_type, printer
+from collections import OrderedDict
+from os import mkdir
+from os.path import exists
+from bs4 import BeautifulSoup
+import urllib.request
+import json
+import csv
+
+
+EDGAR_BASE_URL = "http://www.sec.gov/cgi-bin/browse-edgar?action=getcompany&CIK="
+
+IEX_NUMBER_OF_SYMBOLS_PER_REQUEST = 100
+IEX_BASE_URL = "https://api.iextrading.com/1.0/stock/market/batch"
+IEX_TYPES_FIELDNAMES_TITLES = OrderedDict({
+    "quote": [
+        ["close", "Price"],
+        ["peRatio", "Price/Earnings"]
+    ],
+    "stats": [
+        ["dividendYield", "Dividend Yield"],
+        ["latestEPS", "Earnings/Share"],
+        ["week52high", "52 Week Low"],
+        ["week52low", "52 Week High"],
+        ["marketcap", "Market Cap"],
+        ["EBITDA", "EBITDA"],
+        ["priceToSales", "Price/Sales"],
+        ["priceToBook", "Price/Book"]
+    ]
+})
+
+
+def rename(package: PackageWrapper):
+    package.pkg.descriptor['resources'][0]['name'] = 's-p-constituents'
+    package.pkg.descriptor['resources'][0]['path'] = 'data/constituents.csv'
+    package.pkg.descriptor['resources'][1]['name'] = 'constiuents-financials'
+    package.pkg.descriptor['resources'][1]['path'] = 'data/constiuents-financials.csv'
+    yield package.pkg
+    res_iter = iter(package)
+    first: ResourceWrapper = next(res_iter)
+    second: ResourceWrapper = next(res_iter)
+    yield first.it
+    yield second.it
+    yield from package
+
+
+def extract_constituents(html):
+    datadir = 'data'
+    if not exists(datadir):
+        mkdir(datadir)
+    soup = BeautifulSoup(html, 'html.parser')
+    table = soup.find("table", {"class": "wikitable sortable"})
+    # Fail now if we haven't found the right table
+    header = table.findAll('th')
+    if header[0].string != "Ticker symbol" or header[1].string != "Security":
+        raise Exception("Can't parse wikipedia's table!")
+
+    # Retreive the values in the table
+    rows = table.findAll('tr')
+    for row in rows:
+        fields = row.findAll('td')
+        if fields:
+            yield {
+                'Symbol': fields[0].text,
+                'Name': fields[1].text,
+                'Sector': fields[3].text
+            }
+
+
+# helper function to append column data for each iex type and field pair
+def append_column_data(rows, row_number, data_dict):
+    for iex_type, field in IEX_TYPES_FIELDNAMES_TITLES.items():
+        for field_name in field:
+            rows[row_number].append(data_dict[iex_type][field_name[0]])
+
+
+def extract_financial_data():
+    # get constituents data
+    reader = csv.reader(open('data/constituents.csv'))
```

Comment:
from where `data/constituents.csv` is coming from?

Reply:
From wikipedia at https://en.wikipedia.org/wiki/List_of_S%26P_500_companies. There might be a better way of doing this.

Reply:
Yes, but how did it get there?